cmd/networkcmd/start.go

```diff
@@ -95,154 +109,184 @@ func StartNetwork(*cobra.Command, []string) error {
 	}
 
 	if bootstrapped {
-		if !needsRestart {
-			ux.Logger.PrintToUser("Network has already been booted.")
-			return nil
-		}
-		if _, err := cli.Stop(ctx); err != nil {
-			return err
-		}
-		if err := app.ResetPluginsDir(); err != nil {
-			return err
-		}
+		ux.Logger.PrintToUser("Network has already been booted.")
+		return nil
 	}
 
-	var startMsg string
-	if snapshotName == constants.DefaultSnapshotName {
-		startMsg = "Starting previously deployed and stopped snapshot"
-	} else {
-		startMsg = fmt.Sprintf("Starting previously deployed and stopped snapshot %s...", snapshotName)
+	// this takes about 1 secs
+	avalancheGoBinPath, err := sd.SetupLocalEnv()
+	if err != nil {
+		return err
 	}
-	ux.Logger.PrintToUser(startMsg)
 
 	autoSave := app.Conf.GetConfigBoolValue(constants.ConfigSnapshotsAutoSaveKey)
 
 	tmpDir, err := anrutils.MkDirWithTimestamp(filepath.Join(app.GetRunDir(), "network"))
 	if err != nil {
 		return err
 	}
-
 	rootDir := ""
 	logDir := ""
-	if !autoSave {
-		rootDir = tmpDir
-	} else {
-		logDir = tmpDir
-	}
-
 	pluginDir := app.GetPluginsDir()
-
-	loadSnapshotOpts := []client.OpOption{
-		client.WithExecPath(avalancheGoBinPath),
-		client.WithRootDataDir(rootDir),
-		client.WithLogRootDir(logDir),
-		client.WithReassignPortsIfUsed(true),
-		client.WithPluginDir(pluginDir),
-	}
-
-	// load global node configs if they exist
-	configStr, err := app.Conf.LoadNodeConfig()
+	nodeConfig, err := app.Conf.LoadNodeConfig()
 	if err != nil {
 		return err
 	}
-	if configStr != "" {
-		loadSnapshotOpts = append(loadSnapshotOpts, client.WithGlobalNodeConfig(configStr))
+	if nodeConfig == "" {
+		nodeConfig = "{}"
 	}
-
-	ux.Logger.PrintToUser("Booting Network. Wait until healthy...")
-	resp, err := cli.LoadSnapshot(
-		ctx,
-		snapshotName,
-		app.Conf.GetConfigBoolValue(constants.ConfigSnapshotsAutoSaveKey),
-		loadSnapshotOpts...,
-	)
+	nodeConfig, err = utils.SetJSONKey(nodeConfig, config.ProposerVMUseCurrentHeightKey, true)
 	if err != nil {
-		return fmt.Errorf("failed to start network with the persisted snapshot: %w", err)
+		return err
+	}
+	if flags.SnapshotName == "" {
+		flags.SnapshotName = constants.DefaultSnapshotName
 	}
 
-	ux.Logger.PrintToUser("Node logs directory: %s/node<i>/logs", resp.ClusterInfo.LogRootDir)
-	ux.Logger.PrintToUser("")
-	ux.Logger.PrintToUser("Network ready to use.")
-	ux.Logger.PrintToUser("")
+	snapshotPath := filepath.Join(app.GetSnapshotsDir(), "anr-snapshot-"+flags.SnapshotName)
+	if utils.DirectoryExists(snapshotPath) {
+		var startMsg string
+		if flags.SnapshotName == constants.DefaultSnapshotName {
```

Comment:
nit: not worth displaying snapshot name for the user if it was provided by the user anyway.
so we don't need this if else statement